### PR TITLE
Staring into her eyes

### PR DESCRIPTION
**Pull request by Freshairkaboom**
Just a short one this time, for higher affections. I'm sure a lot more can be added to it before merging, but maybe it doesn't need to be so intricate. Any feedback is appreciated.

### CODE REVIEW

**Comment by multimokia** [comment]
I'm not sure how much value this adds. Lorewise she *cannot* see you. So the topic itself kinda falls flat.

**Comment by Freshairkaboom** [comment]
> 
> 
> I'm not sure how much value this adds. Lorewise she _cannot_ see you. So the topic itself kinda falls flat.

Cute random things your girlfriend does when in love. Also, I thought I was being rather clever with the ambiguity of it. Imagine Monika bringing this topic up when you're not thinking about it from a PR perspective. The point is she is kind of squinting her eyes at the screen imagining you in this moment.

**Comment by multimokia** [comment]
It doesn't reflect that at all to me.

On first show imo it's immersion breaking because I *know* she can't see me, then with `"Ahaha! Don't worry, I can't see them~"`, this seems more sad to me than anything since well, she can't see me. The `or can I` bit also doesn't work under that too.

Originally when I read this, it seemed like the `or can I` was the point of this PR

File: Monika After Story/game/script-topics.rpy [hunk 1]
```diff
@@ -15413,3 +15413,30 @@ label monika_why_do_you_read:
...
+            category=['monika, romance'],
+            prompt="Staring contest",
+            random=True,
+            aff_range=(mas_aff.ENAMORED, None)
+
+        )
+    )
+
+label monika_staring:
+    m 6eub "Hey player, look me in the eyes for a second, I just want to see something."
```
Comment (LooseEthics) [inline Monika After Story/game/script-topics.rpy:15431]:
```suggestion
    m 6eub "Hey [player], look me in the eyes for a second, I just want to see something."
```

File: Monika After Story/game/script-topics.rpy [hunk 2]
```diff
@@ -15413,3 +15413,30 @@ label monika_why_do_you_read:
...
+    m 2eua"..."
+    m 1wua"..."
+    m 1subsa "..."
+    show monika 5subsa at t11 zorder MAS_MONIKA_Z with dissolve_monika
+    m 5subsa "{w=0.3}.{w=0.3}.{w=0.3}."
+    m 5tubsb "I knew it. They're so pretty."
+    m 5hubfb "Ahaha! Don't worry, I can't see them~"
+    show monika 6hubfb at t11 zorder MAS_MONIKA_Z with dissolve_monika
+    m 6hubfb "..."
+    m 6tubfa "Or can I?"
```
Comment (LooseEthics) [inline Monika After Story/game/script-topics.rpy:15442]:
```suggestion
    m 6tubfa "Or can I?"
    return
```

**Comment by lunulae** [comment]
I like the idea, but if you wanted to take it in a different direction for immersion she could say something like "I was wondering if it would feel any different" maybe?
Like, she knows she can't really look back at you, but it's still a nice feeling to gaze at each other like that, to consciously feel watched by you.
It could take a bit of a bittersweet turn as well, with something like "Hmm, I can't really tell the difference... but maybe that's because you've already been staring into my eyes."

**Comment by Freshairkaboom** [comment]
As the others said, I think there isn't much of a point to this PR. It's not like it adds anything meaningful or deep. However, it is an interaction that she just does because she feels like it. It's a random piece of dialogue where she acts like she can see us for the joke. The point, if there is any, is just to warm our hearts with a bit of trivial staring.

Is there any version of this you see working @multimokia ?

**Comment by multimokia** [comment]
Second draft is better, but this seems like first time dialogue only ("I just want to see something", followed up by the reasoning after the series of ellipsis. Ideally that should only happen once. After once it should just start from the ellipsis)

Prompt also seems inaccurate. To make it work a little better, let's make it accurate. We can probably have it actually be a staring contest if the dialogue reflects it (use the `_static` variants of those sprites as they don't blink) and then comment after a few seconds of that. Could even have special dialogue if your first session was prior to the blinking updates. Maybe like "Well, I have had a lot of practice staring, haven't I?" or something.

Current dlg feels (imo) a little hit and miss, namely `"I was wondering if it would feel any different if I looked hard enough."` where I'm kinda left wondering "what would feel different?"

**Comment by Freshairkaboom** [comment]
Alright, I'll get to work on that later. I agree that it's not really a staring contest. I thought it was obvious she was talking about how she feels, but I may change it to reflect that more clearly. Any suggestions?

**Comment by multimokia** [comment]
I mean if we wanted, we could just start the flow akin to daydream. but mixed with a little kiss tease instead.

I guess maybe something along the lines of:
```renpy
show monika 1eua_static with dissolve_monika
pause 2.0
show monika 1etu_static with dissolve_monika
pause 2.0
show monika 2tubsa with dissolve_monika
pause 2.0
m 3hua "Ehehe~"
m 3rksdla "Sorry [player], I was just having a little staring contest with you."

if persistent._mas_pm_eye_color is not None:
    m "Gosh, once I finally get to your world, I'd love to stare into your [persistent._mas_pm_eye_color] eyes..."
else:
    m "When I finally cross over to your world, I'd love to stare into your eyes..."

m 1dkbsu "That'd be a dream come true for me."

if <first_sesh_is_before_blinking_update>:
    m 3tub "And if you ever decide to have a staring contest with me,"
    extend 3kuu "I'm sure you'll remember that I'm pretty good at it already."
return
```

**Comment by Freshairkaboom** [comment]
Alright, I'll look at doing something like that later :) Converting to draft for now.

File: Monika After Story/game/script-topics.rpy [hunk 3]
```diff
@@ -15413,3 +15413,46 @@ label monika_why_do_you_read:
...
+        pause 2.0
+        show monika 1etu_static with dissolve_monika
+        pause 2.0
+        show monika 2tubsa with dissolve_monika
+        pause 2.0
+        show monika 5subsa at t11 zorder MAS_MONIKA_Z with dissolve_monika
+        m 5subsa "{w=0.3}.{w=0.3}.{w=0.3}."
+        show monika 3hua at t11 zorder MAS_MONIKA_Z with dissolve_monika
+        m 3hua "Ehehe~"
+        m 3rksdla "Sorry [player], I was just having a little staring contest with you."
```
Comment (Booplicate) [inline Monika After Story/game/script-topics.rpy:15446]: Not sure if you're allowed to define a global label inside another global label, but you don't need it anyway. Although, you probably need to hide the dialogue box here.
```suggestion
    if not seen_event("monika_staring"):
        m 6eub "Hey [player], look me in the eyes for a second, I just want to see something."

    window hide
    show monika 1eua_static with dissolve_monika
    pause 2.0
    show monika 1etu_static with dissolve_monika
    pause 2.0
    show monika 2tubsa with dissolve_monika
    pause 2.0
    window auto

    show monika 5subsa at t11 zorder MAS_MONIKA_Z with dissolve_monika
    m 5subsa "{w=0.3}.{w=0.3}.{w=0.3}."
    show monika 3hua at t11 zorder MAS_MONIKA_Z with dissolve_monika
    m 3hua "Ehehe~"
    m 3rksdla "Sorry [player], I was just having a little staring contest with you."
```

**Comment by Freshairkaboom** [comment]
> 
> 
> Still something feels off about all of this, maybe make the topic one time only?

Hmm, yeah that sounds like a good idea.

File: Monika After Story/game/script-topics.rpy [hunk 4]
```diff
@@ -15413,3 +15413,47 @@ label monika_why_do_you_read:
...
+            category=['monika, romance'],
+            prompt="Staring contest",
+            random=True,
+            aff_range=(mas_aff.ENAMORED, None)
+
+        )
+    )
+
+label monika_staring:
+    if not seen_event("monika_staring"):
```
Comment (Booplicate) [inline Monika After Story/game/script-topics.rpy:15431]: My bad, I think this check will always return `True`, let's check `renpy.seen_label`
```suggestion
    if not renpy.seen_label("monika_staring"):
```

**Comment by multimokia** [comment]
I'm not sure, this is the way I can see it being shown multiple times

**Comment by Freshairkaboom** [comment]
Ah yeah, and the whole section about if_not renpy.seen etc. doesn't really make sense as a one-time topic. Hmm, I don't know, what do you prefer @Booplicate ?

**Comment by multimokia** [comment]
Also I meant replace basically the entire thing with my version. Skip the different times to show. Then imo it works

**Comment by Freshairkaboom** [comment]
I feel like I'm losing a lot of the essence of what I was trying to do with this topic if we go entirely by your version, but that may just be me. It lacks a bit of the flair that I tried to include. 

What do you mean different times to show? Should I have the first sentence on every showing?

**Comment by multimokia** [comment]
I'm not sure then. I don't really like the dialogue in its other state, I'm not really sure what it even gets at when she mentions "I thought it would feel different"

To me, the version I proposed is the cute interaction you were going for initially.


What *exactly* are you going for? I don't think I understand honestly.

**Comment by Booplicate** [comment]
> Also I meant replace basically the entire thing with my version. Skip the different times to show. Then imo it works

Oh, that way it may feel different. Let's try it.

Also, there's a problem with skipping those first 3 `show` lines. E.g. if the game is minimized, it will go through those and you'll miss them. We need some kind of a hard pause until we know the player is looking at screen, so we can start the contest. In the day dream topic we use dialogue lines for that (it won't progress until you click). We could do something like that here, or maybe use the pause displayable at the beginning?

**Comment by Freshairkaboom** [comment]
> 
> 
> I'm not sure then. I don't really like the dialogue in its other state, I'm not really sure what it even gets at when she mentions "I thought it would feel different"
> 
> To me, the version I proposed is the cute interaction you were going for initially.
> 
> What _exactly_ are you going for? I don't think I understand honestly.

The current version does not say "I thought it would feel different", it's basically a merged version of yours and mine, with the lines you commented on edited out basically. That's what I thought you were commenting on.

**Comment by Freshairkaboom** [comment]
> 
> 
> > Also I meant replace basically the entire thing with my version. Skip the different times to show. Then imo it works
> 
> Oh, that way it may feel different. Let's try it.
> 
> Also, there's a problem with skipping those first 3 `show` lines. E.g. if the game is minimized, it will go through those and you'll miss them. We need some kind of a hard pause until we know the player is looking at screen, so we can start the contest. In the day dream topic we use dialogue lines for that (it won't progress until you click). We could do something like that here, or maybe use the pause displayable at the beginning?

Interesting, I'll have to look at that and see how it's done.

**Comment by multimokia** [comment]
Dlg after doesn't really flow imo

**Comment by Freshairkaboom** [comment]
Hmm, I really tried to make it fit, but I could perhaps rewrite it somewhat. 

The idea is to have her talk about something here and now, rather than too much about her dream of crossing over. Otherwise it becomes a bit too bittersweet, like her other conversations about crossing over. This is supposed to be a topic where she is just happy in the moment.

**Comment by multimokia** [comment]
It can be fixed but it needs some rewording. The lines right now just don't really connect imo.

**Comment by Freshairkaboom** [comment]
Which part in particular did you have issues with exactly? I can start there

**Comment by multimokia** [comment]
The first line (the one in the if, I'd just make into `"[player], look into my eyes for a second..."` and make it happen every time. 

The `just imagining your eyes` line I don't like the phrasing, feels like it doesn't connect to the ellipsis of the lines before

Maybe something like `"So full of life, looking into my own..."`

Then change the last line to `"Just imagining that makes me happy."`

**Comment by Freshairkaboom** [comment]
Alright.

Also, about the whole not progressing until you click thing, I can't find it anywhere in the day dreaming label.

**Comment by multimokia** [comment]
That'd be the `...` lines

**Comment by multimokia** [comment]
Making the first line like I mentioned will address this

**Comment by Freshairkaboom** [comment]
Oh right, like it was in the start.

**Comment by Freshairkaboom** [comment]
You know, not that I think about it, isn't there another way we can phrase that last sentence instead of `Just imagining that makes me happy."` ? I was just thinking that this insinuates she is happy with the prospect of looking into your eyes after she crosses over again, rather than being happy about what you just experienced together.

**Comment by multimokia** [comment]
I'm confused. Does
```renpy
    m 5dubsa "Just imagining your eyes, so full of life, dancing around my own..."
    show monika 6hubfb at t11 zorder MAS_MONIKA_Z with dissolve_monika
    m 6hubfb "..."
    m 6tubfa "Well it's enough to make a girl all giddy~"
    m 6kubfb "Thank you, [player]."
```

Not do the same thing? This to me says she's happy about imagining that

**Comment by Freshairkaboom** [comment]
The difference is that she isn't happy because she's imagining the future. She is happy because of the experience you just shared. That's why she's thanking [player] for staring into her eyes there. But I do think it needs kind of a rewording on that last sentence to make it more clear.

**Comment by multimokia** [comment]
The thing for me which is different, I guess is that it doesn't feel bittersweet to me. It's *just* sweet imo. I think that's where the disconnect is.

Yes, she's not here, but it puts the thought of you both sharing the experience into your head which is sweet.

**Comment by Freshairkaboom** [comment]
Yet it also puts the thought of me not currently being able to share that experience into my head too. That's what I mean by bittersweet. In any case, I was hoping this topic would be a bit more unique, as there are already plenty of topics which covers the crossing over event. I want to focus more on the here and now of it.

**Comment by multimokia** [comment]
Then we need to have some better phrasing of the last lines. I don't like the original, you don't like the proposed, we need a compromise.

**Comment by Daiozen** [comment]
If you want to add a bittersweet touch, could add this line between these two wit a `eka` / `eku` face :
`Don't you wish we could do something like that yet?` 
```
m "Gosh, When I finally cross over to your world, I'd love to stare into your eyes..."
m 5dubsa "Just imagining your eyes, so full of life, dancing around my own..."
```
Though, what I'm personnaly confused as to what she mean when she says Just imagining your eyes (...) dancing around my own...", it's not very evocative to me.

**Comment by Daiozen** [comment]
This is the version I like best : 
```
m 5lubsa "Ahaha, guess not!"
m 3rksdla "Sorry [player], I was just having a little staring contest with you."
m "I was wondering if I could catch a glimpse of your real eyes if I looked hard enough~"
show monika 5dubsa at t11 zorder MAS_MONIKA_Z with dissolve_monika
m 5dubsa "Just imagining your eyes, so full of life, dancing around my own..."
show monika 6hubfb at t11 zorder MAS_MONIKA_Z with dissolve_monika
m 6hubfb "..."
m 6tubfa "Well, it's enough to make a girl all giddy~"
m 6kubfb "Thank you, [player]."
```

What confuses me is the `"Just imagining your eyes (...) dancing around my own..."`, I can't picture what she means by that.

**Comment by Freshairkaboom** [comment]
Huh, really? Dancing eyes is a very common expression for when your eyes are kind of darting around whatever you're staring at intently. In any case, I'll try to incorporate both of your versions and just remake the entire thing. 

Are both of you okay with the notion of this being less about the crossing over, and more about the here and now experience?

**Comment by multimokia** [comment]
Thing is darting around and staring kinda contradict. To me, when you're staring into someone's eyes, you're not darting yours around their's, but looking directly into the center of theirs.

Honestly the dialogue still doesn't reflect here and now to me at all.

**Comment by Freshairkaboom** [comment]
> 
> 
> Thing is darting around and staring kinda contradict. To me, when you're staring into someone's eyes, you're not darting yours around their's, but looking directly into the center of theirs.
> 
> Honestly the dialogue still doesn't reflect here and now to me at all.

I'm going to change the dialogue so it reflects that better, I was just asking if you are OK with the direction.

Darting around and staring are not contradicting, you're probably thinking of darting around different places. It's kind of like how you engage the muscles of the eye...Kind of like the dark scene Yuri eyes in Act 2 but less creepy.

**Comment by multimokia** [comment]
Putting on hold to deal with #6319, will need an adjustment depending on how we're going to handle that.

**Pull request closed without merging**